lib/matplotlib/scale.py

```diff
@@ -82,6 +82,88 @@ def get_transform(self):
         return IdentityTransform()
 
 
+class ArbitraryTransform(Transform):
+     """
+     A simple transform that takes and arbitrary function for the
+     forward and inverse transform.
+     """
+
+     input_dims = 1
+     output_dims = 1
+     is_separable = True
+     has_inverse = True
+
+     def __init__(self, forward, inverse):
+         """
+         Parameters
+         ----------
+
+         forward: The forward function for the transform
+
+         inverse: The inverse of the forward function.
+         """
+         super().__init__()
+         if callable(forward) and callable(inverse):
+             self._forward = forward
+             self._inverse = inverse
+         else:
+             raise ValueError('arguments to ArbitraryTransform must '
+                              'be functions.')
+
+     def transform_non_affine(self, values):
+         with np.errstate(divide='ignore', invalid='ignore'):
+             q = self._forward(values)
+         return q
+
+     def inverted(self):
+         return ArbitraryTransform(self._inverse, self._forward)
+
+
+class ArbitraryScale(ScaleBase):
+
+     name = 'arbitrary'
+
+     def __init__(self, axis, functions=None):
```

Comment:
Just don't set a default to `functions`?

Reply:
Thats a good idea!

Reply:
So the problem w/ passing the class instead of an instance of the class is that the Locators and Formatters need to be passed arguments, so I think it makes more sense to do 
```
     def __init__(self, axis, functions, *,
                  major_locator=AutoLocator(),
                  minor_locator=NullLocator(),
                  major_formatter=ScalarFormatter(),
                  minor_formatter=NullFormatter()):
```

Reply:
But that won't work because you can't share a locator/formatter across multiple axes right now (as they keep a reference to their axis -- they probably shouldn't, but that's another in depth refactoring...).

Reply:
Hmmm, well, I guess folks will just have to change the locators and formatters manually